/*
 *  Copyright (c) 2016, https://github.com/wubenqi
 *  All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package zproto;

option java_package = "com.zchat.engine.zproto";
option java_outer_classname = "ZProtoPresences";
option optimize_for = CODE_SIZE;

// Presences

message ZUserID {
  uint32 app_id = 1;
  string user_id = 2;
};

message ZOnlineUser {
  uint32 app_id       = 1;
  string user_id      = 2;
  uint32 server_id    = 3;
  uint64 conn_id      = 4;
}

///////////////////////////////////////////////////////////////////////
// online
// ClientOnlineReq -> ClientOnlineRsp
message ClientOnlineReq {
  uint32 server_id = 1;
  uint64 conn_id = 2;
  uint32 app_id = 3;
  string user_id = 4;
  uint32 state = 5;
}

// ClientOnlineRsp
message ClientOnlineRsp {
  uint64 index_id = 1;
}

// ClientOfflineReq -> ClientOfflineRsp
message ClientOfflineReq {
  uint32 server_id = 1;
  uint64 conn_id = 2;
}

// ClientOfflineRsp
message ClientOfflineRsp {
  uint64 index_id = 1;
}

// QueryOnlineUserReq -> QueryOnlineUserRsp
message QueryOnlineUserReq {
  repeated ZUserID user_id_list = 1;
}

// QueryOnlineUserRsp
message QueryOnlineUserRsp {
  repeated ZOnlineUser online_users = 1;
}
